lib/src/main/java/com/auth0/jwt/algorithms/Algorithm.java

```diff
@@ -367,14 +369,32 @@ public String toString() {
      * @throws SignatureGenerationException if the Key is invalid.
      */
     public byte[] sign(byte[] headerBytes, byte[] payloadBytes) throws SignatureGenerationException {
+        PrivateKey privateKey = getPrivateKeyDetails() == null ? null : getPrivateKeyDetails().getPrivateKey();
+        return sign(headerBytes, payloadBytes, privateKey);
+    }
+
+    /**
+     * Sign the given content using this Algorithm instance.
+     *
+     * @param headerBytes  an array of bytes representing the base64 encoded header content
+     *                     to be verified against the signature.
+     * @param payloadBytes an array of bytes representing the base64 encoded payload content
+     *                     to be verified against the signature.
+     * @param privateKey   A private key with which the content has to be signed
+     *
+     * @return the signature in a base64 encoded array of bytes
+     * @throws SignatureGenerationException if the Key is invalid.
+     */
+    public byte[] sign(byte[] headerBytes, byte[] payloadBytes, PrivateKey privateKey)
```

Comment:
We are exposing this new method in Algorithm class which takes in a Private Key, this is to ensure the same Private key used to build the header claim is used for signing. But this cannot be achieved without making this method public. This takes in a Private Key where Algorithm already has a private key which is contradictory